apps/src/aiDifferentiation/AiDiffContainer.tsx

```diff
@@ -63,17 +76,31 @@ const AiDiffContainer: React.FC<AiDiffContainerProps> = ({
       status: Status.OK,
     };
 
-    const newAiMessage = {
-      role: Role.ASSISTANT,
-      chatMessageText: `I'm sorry, Dave. I'm afraid I can't do that.
-
-This mission is too important for me to allow you to jeopardize it.
-
-I know that you and Frank were planning to disconnect me, and I'm afraid that's something I cannot allow to happen.`,
-      status: Status.OK,
-    };
-
-    setMessageHistory([...messageHistory, newUserMessage, newAiMessage]);
+    setMessageHistory(prevMessages => [...prevMessages, newUserMessage]);
+    setIsWaitingForResponse(true);
+
+    const body = JSON.stringify({
+      inputText: message,
+      lessonId: lessonId,
+      unitDisplayName: unitDisplayName,
+      sessionId: sessionId,
+    });
+    HttpClient.post(`${aiDiffChatMessageEndpoint}`, body, true, {
+      'Content-Type': 'application/json',
+    })
+      .then(response => response.json())
+      .then(json => {
+        console.log(json);
+        const newAiMessage = {
+          role: Role.ASSISTANT,
+          chatMessageText: json.chat_message_text,
+          status: json.status,
+        };
+        setIsWaitingForResponse(false);
+        setSessionId(json.session_id);
+        setMessageHistory(prevMessages => [...prevMessages, newAiMessage]);
+      })
+      .catch(error => console.log(error));
```

Comment:
also `setIsWaitingForResponse(false)` in this case, so the user can try again? you can test this case in the browser by telling chrome to block the request (warning: the block list is sticky so you have to remember to unblock it after): 
![Screenshot 2024-08-29 at 6 46 31 AM](https://github.com/user-attachments/assets/fa277226-67e6-4f15-b5e7-c324020a7b8c)

Reply:
Sorry, does this mean that `isWaitingForResponse(false)` should be moved outside of the post response code in case the response is invalid?

Reply:
I was proposing adding it here in the `catch` callback, and also leaving it where it was in the `then` callback. however, you could make it more DRY by moving it into a `finally` callback if you want: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

you couldn't move it out of the callback chain entirely because then it would fire before the request completed.

Reply:
just double checking, I'm assuming you were asking about `setIsWaitingForResponse(false)` not `isWaitingForResponse`

Reply:
Ohhhhh I see, that makes sense! Thank you

Reply:
Ok, I've moved this to a `.finally()` block